spark/dl/src/main/scala/org/apache/spark/ml/DLEstimator.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.ml
+import com.intel.analytics.bigdl.dataset.{DataSet, MiniBatch}
+import com.intel.analytics.bigdl.{Criterion, Module}
+import com.intel.analytics.bigdl.optim.Optimizer
+import com.intel.analytics.bigdl.tensor.{Tensor}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import org.apache.spark.ml.param.shared.HasInputCols
+import org.apache.spark.ml.param.{Param, ParamMap, Params}
+import org.apache.spark.sql.{DataFrame}
+import org.apache.spark.rdd.RDD
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+/**
+ * A wrapper of Optimizer to support fit() in ML Pipelines as an Estimator
+ */
+class DLEstimator[@specialized(Float, Double) T: ClassTag]
+(override val uid: String = "DLEstimator")(implicit ev: TensorNumeric[T])
+  extends MLEstimator with HasInputCols with DLDataParams[T]{
```

Comment:
I didn't know these traits before,Actually it makes more sense to use this traits together with other inputs (feature and label size) as params